Editor/Overlays/CinemachineToolbarOverlay.cs

```diff
@@ -1,138 +1,169 @@
 #if UNITY_2021_2_OR_NEWER
+using System;
 using UnityEditor;
+using UnityEditor.EditorTools;
 using UnityEditor.Overlays;
 using UnityEditor.Toolbars;
 using UnityEngine;
 using UnityEngine.UIElements;
+#if UNITY_2022_1_OR_NEWER
+using System.Collections.Generic;
+#endif
 
 namespace Cinemachine.Editor
 {
     /// <summary>
-    /// To display a CinemachineExclusiveEditorToolbarToggle in the Cinemachine Toolbar,
-    /// TODO: Make this extendable when another PR lands (see: CMCL-501),
+    /// This is a generic Tool class for Cinemachine tools.
+    /// To create a new tool, inherit from this class and implement OnEnable to set m_IconContent's image and tooltip
+    /// attributes. If you implement a constructor for your tool, don't forget to call the base constructor.
+    ///
+    /// A tool will be drawn iff it has been registered using CinemachineSceneToolUtility.RegisterTool.
+    /// This is generally done in the OnEnable function of the editor script of the cinemachine component
+    /// (CinemachineVirtualCamera, CinemachineComponentBase), for which the tool was meant.
+    /// To unregister, call CinemachineSceneToolUtility.UnregisterTool in the same script's OnDisable function.
+    ///
+    /// To draw the handles related to the tool, you need to implement your drawing function and call it in the
+    /// editor script's OnSceneGUI function. An alternative for drawing handles is to override this function's
+    /// OnToolGUI or OnDrawHandles functions (see EditorTool or IDrawSelectedHandles docs for more information).
+    ///
+    /// To check, if a tool has been enabled/disabled in the editor script, use CinemachineSceneToolUtility.IsToolActive.
     /// </summary>
-    [Overlay(typeof(SceneView), "Cinemachine")]
-    [Icon("Packages/com.unity.cinemachine/Gizmos/cm_logo.png")]
-    class CinemachineToolbarOverlay : ToolbarOverlay
+    public class CinemachineTool : EditorTool, IDrawSelectedHandles
```

Comment:
It is extendable, because the Tools bar is extendable. That's why I made it public. Is there a reason for making it private?

Reply:
But it's only extendable if UNITY_2022_1_OR_NEWER

Reply:
No, that was the case before when we had a separate toolbar (it uses different methods/classes). The Tools toolbar is special, it is extended by this attribute:
[EditorTool("Field of View Tool", typeof(CinemachineVirtualCameraBase))]

But because of the above issue, I need to bring back the old toolbar anyway, so this will be indeed in UNITY_2022_1_OR_NEWER :S